wrapper/src/main/java/software/amazon/jdbc/util/ConnectionUrlParser.java

```diff
@@ -188,6 +192,42 @@ public static void parsePropertiesFromUrl(final String url, final Properties pro
       final String currentParameterName = param.substring(0, pos);
       props.setProperty(currentParameterName, currentParameterValue);
     }
+
+    // try to detect "wrapperCaseSensitive" parameter
+    String caseSensitiveSettingStr = props.getProperty(PropertyDefinition.CASE_SENSITIVE.name);
+    if (caseSensitiveSettingStr == null) {
+      // try low case
+      caseSensitiveSettingStr = props.getProperty(PropertyDefinition.CASE_SENSITIVE.name.toLowerCase());
+    }
+    if (caseSensitiveSettingStr == null) {
+      caseSensitiveSettingStr = PropertyDefinition.CASE_SENSITIVE.defaultValue;
+    }
+
+    final boolean caseSensitive = Boolean.parseBoolean(caseSensitiveSettingStr);
+
+    if (!caseSensitive) {
+      // restore case for parsed parameters
+      final Properties propsCopy = new Properties();
+      propsCopy.putAll(props);
+      props.clear();
+      for (final Map.Entry<Object, Object> entry : propsCopy.entrySet()) {
+        AwsWrapperProperty awsWrapperProperty = PropertyDefinition.byName(entry.getKey().toString());
+        if (awsWrapperProperty != null) {
+          props.setProperty(entry.getKey().toString(), entry.getValue().toString());
+          continue;
+        }
+
+        awsWrapperProperty = PropertyDefinition.byNameIgnoreCase(entry.getKey().toString());
+        if (awsWrapperProperty != null) {
+          // use original (camel-case) property name
+          props.setProperty(awsWrapperProperty.name, entry.getValue().toString());
+          continue;
+        }
```

Comment:
I think we can simplify this to:

```suggestion
        final AwsWrapperProperty awsWrapperProperty = PropertyDefinition.byName(entry.getKey().toString(), true);
        if (awsWrapperProperty != null) {
          props.setProperty(awsWrapperProperty.name, entry.getValue().toString());
          continue;
        }
```